server: TestInitHandshake failed

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3195690&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3195690&tab=artifacts#/) on master @ [24825aaab7700178b01efc3b7997d3511a3cf22b](https://github.com/cockroachdb/cockroach/commits/24825aaab7700178b01efc3b7997d3511a3cf22b):


```
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:257 +0x292
        github.com/cockroachdb/cockroach/pkg/server.enhanceHandlerContextWithHTTPClient.func1(0x54d89f0, 0xc000010010, 0xc003b5c300)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:226 +0xe2
        net/http.HandlerFunc.ServeHTTP(0xc006510000, 0x54d89f0, 0xc000010010, 0xc003b5c300)
        	/usr/local/go/src/net/http/server.go:2049 +0x44
        net/http.(*ServeMux).ServeHTTP(0xc007474440, 0x54d89f0, 0xc000010010, 0xc003b5c300)
        	/usr/local/go/src/net/http/server.go:2428 +0x1ad
        net/http.serverHandler.ServeHTTP(0xc00848e8c0, 0x54d89f0, 0xc000010010, 0xc003b5c300)
        	/usr/local/go/src/net/http/server.go:2867 +0xa3
        net/http.initALPNRequest.ServeHTTP(0x54f6818, 0xc009a52210, 0xc009a08700, 0xc00848e8c0, 0x54d89f0, 0xc000010010, 0xc003b5c300)
        	/usr/local/go/src/net/http/server.go:3439 +0x8d
        net/http.(*http2serverConn).runHandler(0xc0048e6d80, 0xc000010010, 0xc003b5c300, 0xc00657e150)
        	/usr/local/go/src/net/http/h2_bundle.go:5723 +0x8b
        created by net/http.(*http2serverConn).processHeaders
        	/usr/local/go/src/net/http/h2_bundle.go:5453 +0x505
        Leaked goroutine: goroutine 138089 [chan send, 10 minutes]:
        github.com/cockroachdb/cockroach/pkg/server.(*tlsInitHandshaker).onTrustInit(0xc001f775f0, 0x54f6818, 0xc009e29350, 0x54d89f0, 0xc000010558, 0xc003b5c900)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:257 +0x292
        github.com/cockroachdb/cockroach/pkg/server.enhanceHandlerContextWithHTTPClient.func1(0x54d89f0, 0xc000010558, 0xc003b5c900)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:226 +0xe2
        net/http.HandlerFunc.ServeHTTP(0xc006510000, 0x54d89f0, 0xc000010558, 0xc003b5c900)
        	/usr/local/go/src/net/http/server.go:2049 +0x44
        net/http.(*ServeMux).ServeHTTP(0xc007474440, 0x54d89f0, 0xc000010558, 0xc003b5c900)
        	/usr/local/go/src/net/http/server.go:2428 +0x1ad
        net/http.serverHandler.ServeHTTP(0xc00848e8c0, 0x54d89f0, 0xc000010558, 0xc003b5c900)
        	/usr/local/go/src/net/http/server.go:2867 +0xa3
        net/http.initALPNRequest.ServeHTTP(0x54f6818, 0xc009a52210, 0xc009a08700, 0xc00848e8c0, 0x54d89f0, 0xc000010558, 0xc003b5c900)
        	/usr/local/go/src/net/http/server.go:3439 +0x8d
        net/http.(*http2serverConn).runHandler(0xc0048e6d80, 0xc000010558, 0xc003b5c900, 0xc00657e348)
        	/usr/local/go/src/net/http/h2_bundle.go:5723 +0x8b
        created by net/http.(*http2serverConn).processHeaders
        	/usr/local/go/src/net/http/h2_bundle.go:5453 +0x505
        Leaked goroutine: goroutine 138119 [chan send, 10 minutes]:
        github.com/cockroachdb/cockroach/pkg/server.(*tlsInitHandshaker).onTrustInit(0xc001f775f0, 0x54f6818, 0xc00a6c2540, 0x54d89f0, 0xc00a6ae0a8, 0xc0018dad00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:257 +0x292
        github.com/cockroachdb/cockroach/pkg/server.enhanceHandlerContextWithHTTPClient.func1(0x54d89f0, 0xc00a6ae0a8, 0xc0018dad00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:226 +0xe2
        net/http.HandlerFunc.ServeHTTP(0xc006510000, 0x54d89f0, 0xc00a6ae0a8, 0xc0018dad00)
        	/usr/local/go/src/net/http/server.go:2049 +0x44
        net/http.(*ServeMux).ServeHTTP(0xc007474440, 0x54d89f0, 0xc00a6ae0a8, 0xc0018dad00)
        	/usr/local/go/src/net/http/server.go:2428 +0x1ad
        net/http.serverHandler.ServeHTTP(0xc00848e8c0, 0x54d89f0, 0xc00a6ae0a8, 0xc0018dad00)
        	/usr/local/go/src/net/http/server.go:2867 +0xa3
        net/http.initALPNRequest.ServeHTTP(0x54f6818, 0xc009a52210, 0xc009a08700, 0xc00848e8c0, 0x54d89f0, 0xc00a6ae0a8, 0xc0018dad00)
        	/usr/local/go/src/net/http/server.go:3439 +0x8d
        net/http.(*http2serverConn).runHandler(0xc0048e6d80, 0xc00a6ae0a8, 0xc0018dad00, 0xc005eb2b40)
        	/usr/local/go/src/net/http/h2_bundle.go:5723 +0x8b
        created by net/http.(*http2serverConn).processHeaders
        	/usr/local/go/src/net/http/h2_bundle.go:5453 +0x505
--- FAIL: TestInitHandshake (605.07s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestInitHandshake PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>
/cc @cockroachdb/server itsbilal
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-8705

Epic CRDB-27234

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3247487&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3247487&tab=artifacts#/) on master @ [701b177d8f4b81d8654dfb4090a2cd3cf82e63a7](https://github.com/cockroachdb/cockroach/commits/701b177d8f4b81d8654dfb4090a2cd3cf82e63a7):


```
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  |   | runtime.goexit
        	            	  |   | 	/usr/local/go/src/runtime/asm_amd64.s:1371
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s: error when sending bundle to peers as leader: Post "https://127.0.0.1:40649/deliverBundle/": dial tcp 127.0.0.1:40649: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0s
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:613
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:89
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	/usr/local/go/src/runtime/asm_amd64.s:1371
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:40649/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:40649
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:546
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:89
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestInitHandshake128222768
--- FAIL: TestInitHandshake (600.03s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestInitHandshake PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/server itsbilal
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3407678&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3407678&tab=artifacts#/) on master @ [e369d86b3bac2b9f40d760c3e6e49f55b7e15abe](https://github.com/cockroachdb/cockroach/commits/e369d86b3bac2b9f40d760c3e6e49f55b7e15abe):


```
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:257 +0x292
        github.com/cockroachdb/cockroach/pkg/server.enhanceHandlerContextWithHTTPClient.func1(0x58ef3e0, 0xc009e62000, 0xc00647a100)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:226 +0xe2
        net/http.HandlerFunc.ServeHTTP(0xc0085e3540, 0x58ef3e0, 0xc009e62000, 0xc00647a100)
        	/usr/local/go/src/net/http/server.go:2049 +0x44
        net/http.(*ServeMux).ServeHTTP(0xc005c604c0, 0x58ef3e0, 0xc009e62000, 0xc00647a100)
        	/usr/local/go/src/net/http/server.go:2428 +0x1ad
        net/http.serverHandler.ServeHTTP(0xc004ed27e0, 0x58ef3e0, 0xc009e62000, 0xc00647a100)
        	/usr/local/go/src/net/http/server.go:2867 +0xa3
        net/http.initALPNRequest.ServeHTTP(0x590e2f8, 0xc004e5ff20, 0xc00334a000, 0xc004ed27e0, 0x58ef3e0, 0xc009e62000, 0xc00647a100)
        	/usr/local/go/src/net/http/server.go:3439 +0x8d
        net/http.(*http2serverConn).runHandler(0xc005362f00, 0xc009e62000, 0xc00647a100, 0xc0013f0018)
        	/usr/local/go/src/net/http/h2_bundle.go:5723 +0x8b
        created by net/http.(*http2serverConn).processHeaders
        	/usr/local/go/src/net/http/h2_bundle.go:5453 +0x505
        Leaked goroutine: goroutine 143097 [chan send, 10 minutes]:
        github.com/cockroachdb/cockroach/pkg/server.(*tlsInitHandshaker).onTrustInit(0xc00c191380, 0x590e2f8, 0xc001ff8e70, 0x58ef3e0, 0xc00d838020, 0xc006344800)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:257 +0x292
        github.com/cockroachdb/cockroach/pkg/server.enhanceHandlerContextWithHTTPClient.func1(0x58ef3e0, 0xc00d838020, 0xc006344800)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:226 +0xe2
        net/http.HandlerFunc.ServeHTTP(0xc0085e3540, 0x58ef3e0, 0xc00d838020, 0xc006344800)
        	/usr/local/go/src/net/http/server.go:2049 +0x44
        net/http.(*ServeMux).ServeHTTP(0xc005c604c0, 0x58ef3e0, 0xc00d838020, 0xc006344800)
        	/usr/local/go/src/net/http/server.go:2428 +0x1ad
        net/http.serverHandler.ServeHTTP(0xc004ed27e0, 0x58ef3e0, 0xc00d838020, 0xc006344800)
        	/usr/local/go/src/net/http/server.go:2867 +0xa3
        net/http.initALPNRequest.ServeHTTP(0x590e2f8, 0xc004e5ff20, 0xc00334a000, 0xc004ed27e0, 0x58ef3e0, 0xc00d838020, 0xc006344800)
        	/usr/local/go/src/net/http/server.go:3439 +0x8d
        net/http.(*http2serverConn).runHandler(0xc005362f00, 0xc00d838020, 0xc006344800, 0xc00b1f2168)
        	/usr/local/go/src/net/http/h2_bundle.go:5723 +0x8b
        created by net/http.(*http2serverConn).processHeaders
        	/usr/local/go/src/net/http/h2_bundle.go:5453 +0x505
        Leaked goroutine: goroutine 143126 [chan send, 10 minutes]:
        github.com/cockroachdb/cockroach/pkg/server.(*tlsInitHandshaker).onTrustInit(0xc00c191380, 0x590e2f8, 0xc007a7a270, 0x58ef3e0, 0xc00bb34020, 0xc00b178100)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:257 +0x292
        github.com/cockroachdb/cockroach/pkg/server.enhanceHandlerContextWithHTTPClient.func1(0x58ef3e0, 0xc00bb34020, 0xc00b178100)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:226 +0xe2
        net/http.HandlerFunc.ServeHTTP(0xc0085e3540, 0x58ef3e0, 0xc00bb34020, 0xc00b178100)
        	/usr/local/go/src/net/http/server.go:2049 +0x44
        net/http.(*ServeMux).ServeHTTP(0xc005c604c0, 0x58ef3e0, 0xc00bb34020, 0xc00b178100)
        	/usr/local/go/src/net/http/server.go:2428 +0x1ad
        net/http.serverHandler.ServeHTTP(0xc004ed27e0, 0x58ef3e0, 0xc00bb34020, 0xc00b178100)
        	/usr/local/go/src/net/http/server.go:2867 +0xa3
        net/http.initALPNRequest.ServeHTTP(0x590e2f8, 0xc004e5ff20, 0xc00334a000, 0xc004ed27e0, 0x58ef3e0, 0xc00bb34020, 0xc00b178100)
        	/usr/local/go/src/net/http/server.go:3439 +0x8d
        net/http.(*http2serverConn).runHandler(0xc005362f00, 0xc00bb34020, 0xc00b178100, 0xc0013f0090)
        	/usr/local/go/src/net/http/h2_bundle.go:5723 +0x8b
        created by net/http.(*http2serverConn).processHeaders
        	/usr/local/go/src/net/http/h2_bundle.go:5453 +0x505
--- FAIL: TestInitHandshake (605.03s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestInitHandshake PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/server itsbilal
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3499226&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3499226&tab=artifacts#/) on master @ [ecaa4920192549af1418923648769ede3de1c9c8](https://github.com/cockroachdb/cockroach/commits/ecaa4920192549af1418923648769ede3de1c9c8):


```
    init_handshake_test.go:96: n1: generating cert bundle for cluster
    init_handshake_test.go:96: n1: sending cert bundle to peers
    init_handshake_test.go:96: n2: received cert bundle
    init_handshake_test.go:122: 
        	Error Trace:	init_handshake_test.go:122
        	Error:      	Received unexpected error:
        	            	invalid signature in messages from peers; likely due to token mismatch
        	            	(1) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.002s): error when sending bundle to peers as leader: Post "https://127.0.0.1:40337/deliverBundle/": dial tcp 127.0.0.1:40337: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.002s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:613
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	/usr/local/go/src/runtime/asm_amd64.s:1371
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:40337/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:40337
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:546
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
        	            	Wraps: (3) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestInitHandshake626115018
--- FAIL: TestInitHandshake (600.03s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestInitHandshake PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/server itsbilal
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3687611&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3687611&tab=artifacts#/) on master @ [ed35bbe8f6e5947db6e3f2423d1aa9dde1d15804](https://github.com/cockroachdb/cockroach/commits/ed35bbe8f6e5947db6e3f2423d1aa9dde1d15804):


```
    init_handshake_test.go:96: n1: sending cert bundle to peers
    init_handshake_test.go:96: n3: received cert bundle
    init_handshake_test.go:122: 
        	Error Trace:	init_handshake_test.go:122
        	Error:      	Received unexpected error:
        	            	operation "init handshake" timed out after 10m0s (took 10m0.001s): invalid signature in messages from peers; likely due to token mismatch
        	            	(1) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.003s): error when sending bundle to peers as leader: Post "https://127.0.0.1:45583/deliverBundle/": dial tcp 127.0.0.1:45583: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.003s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:613
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	/usr/local/go/src/runtime/asm_amd64.s:1371
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:45583/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:45583
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) operation "init handshake" timed out after 10m0s (took 10m0.001s)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:546
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *contextutil.TimeoutError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestInitHandshake713332326
--- FAIL: TestInitHandshake (600.03s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3752985&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3752985&tab=artifacts#/) on master @ [1a078671083946342ecf610a6fb899df2d1783d3](https://github.com/cockroachdb/cockroach/commits/1a078671083946342ecf610a6fb899df2d1783d3):


```
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:257 +0x23c
        github.com/cockroachdb/cockroach/pkg/server.enhanceHandlerContextWithHTTPClient.func1({0x515dd80, 0xc008c12000}, 0xc005e2e100)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:226 +0xa3
        net/http.HandlerFunc.ServeHTTP(0xc0047a6140, {0x515dd80, 0xc008c12000}, 0x0)
        	/usr/local/go/src/net/http/server.go:2046 +0x2f
        net/http.(*ServeMux).ServeHTTP(0x0, {0x515dd80, 0xc008c12000}, 0xc005e2e100)
        	/usr/local/go/src/net/http/server.go:2424 +0x149
        net/http.serverHandler.ServeHTTP({0xc005730ab8}, {0x515dd80, 0xc008c12000}, 0xc005e2e100)
        	/usr/local/go/src/net/http/server.go:2878 +0x43b
        net/http.initALPNRequest.ServeHTTP({{0x519a358, 0xc0042b2120}, 0xc0055ee700, {0xc0070342a0}}, {0x515dd80, 0xc008c12000}, 0xc005e2e100)
        	/usr/local/go/src/net/http/server.go:3479 +0x245
        net/http.(*http2serverConn).runHandler(0xc00249afa0, 0xc00249afd0, 0x84b986, 0xc002026600)
        	/usr/local/go/src/net/http/h2_bundle.go:5841 +0x78
        created by net/http.(*http2serverConn).processHeaders
        	/usr/local/go/src/net/http/h2_bundle.go:5571 +0x510
        Leaked goroutine: goroutine 140970 [chan send, 10 minutes]:
        github.com/cockroachdb/cockroach/pkg/server.(*tlsInitHandshaker).onTrustInit(0xc00594c0d0, {0x519a358, 0xc008918840}, {0x515dd80, 0xc002014050}, 0xc00a308500)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:257 +0x23c
        github.com/cockroachdb/cockroach/pkg/server.enhanceHandlerContextWithHTTPClient.func1({0x515dd80, 0xc002014050}, 0xc00a308500)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:226 +0xa3
        net/http.HandlerFunc.ServeHTTP(0xc0047a6140, {0x515dd80, 0xc002014050}, 0x382cf01)
        	/usr/local/go/src/net/http/server.go:2046 +0x2f
        net/http.(*ServeMux).ServeHTTP(0x0, {0x515dd80, 0xc002014050}, 0xc00a308500)
        	/usr/local/go/src/net/http/server.go:2424 +0x149
        net/http.serverHandler.ServeHTTP({0x0}, {0x515dd80, 0xc002014050}, 0xc00a308500)
        	/usr/local/go/src/net/http/server.go:2878 +0x43b
        net/http.initALPNRequest.ServeHTTP({{0x519a358, 0xc0042b2120}, 0xc0055ee700, {0xc0070342a0}}, {0x515dd80, 0xc002014050}, 0xc00a308500)
        	/usr/local/go/src/net/http/server.go:3479 +0x245
        net/http.(*http2serverConn).runHandler(0x5146730, 0x770d098, 0x0, 0x0)
        	/usr/local/go/src/net/http/h2_bundle.go:5841 +0x78
        created by net/http.(*http2serverConn).processHeaders
        	/usr/local/go/src/net/http/h2_bundle.go:5571 +0x510
        Leaked goroutine: goroutine 140997 [chan send, 10 minutes]:
        github.com/cockroachdb/cockroach/pkg/server.(*tlsInitHandshaker).onTrustInit(0xc00594c0d0, {0x519a358, 0xc008ae2d20}, {0x515dd80, 0xc005e46068}, 0xc00320ab00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:257 +0x23c
        github.com/cockroachdb/cockroach/pkg/server.enhanceHandlerContextWithHTTPClient.func1({0x515dd80, 0xc005e46068}, 0xc00320ab00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:226 +0xa3
        net/http.HandlerFunc.ServeHTTP(0xc0047a6140, {0x515dd80, 0xc005e46068}, 0x382cf01)
        	/usr/local/go/src/net/http/server.go:2046 +0x2f
        net/http.(*ServeMux).ServeHTTP(0x0, {0x515dd80, 0xc005e46068}, 0xc00320ab00)
        	/usr/local/go/src/net/http/server.go:2424 +0x149
        net/http.serverHandler.ServeHTTP({0x0}, {0x515dd80, 0xc005e46068}, 0xc00320ab00)
        	/usr/local/go/src/net/http/server.go:2878 +0x43b
        net/http.initALPNRequest.ServeHTTP({{0x519a358, 0xc00a0d9ce0}, 0xc00a3e6700, {0xc0070342a0}}, {0x515dd80, 0xc005e46068}, 0xc00320ab00)
        	/usr/local/go/src/net/http/server.go:3479 +0x245
        net/http.(*http2serverConn).runHandler(0x5146730, 0x770d098, 0x0, 0x0)
        	/usr/local/go/src/net/http/h2_bundle.go:5841 +0x78
        created by net/http.(*http2serverConn).processHeaders
        	/usr/local/go/src/net/http/h2_bundle.go:5571 +0x510
--- FAIL: TestInitHandshake (605.03s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3775215&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3775215&tab=artifacts#/) on master @ [cfb0bf33148744ec06d5d91a1927ca3115101782](https://github.com/cockroachdb/cockroach/commits/cfb0bf33148744ec06d5d91a1927ca3115101782):


```
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:42453/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:42453
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.002s): context canceled before init bundle received from leader
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.002s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:638
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  |   | runtime.goexit
        	            	  |   | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:546
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestInitHandshake2235730890
--- FAIL: TestInitHandshake (600.03s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3861340&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3861340&tab=artifacts#/) on master @ [39923c0b11d229b394fa5498ee58e455cae8ec99](https://github.com/cockroachdb/cockroach/commits/39923c0b11d229b394fa5498ee58e455cae8ec99):


```
    init_handshake_test.go:96: n1: generating cert bundle for cluster
    init_handshake_test.go:96: n1: sending cert bundle to peers
    init_handshake_test.go:96: n3: received cert bundle
    init_handshake_test.go:122: 
        	Error Trace:	init_handshake_test.go:122
        	Error:      	Received unexpected error:
        	            	invalid signature in messages from peers; likely due to token mismatch
        	            	(1) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.001s): error when sending bundle to peers as leader: Post "https://127.0.0.1:41583/deliverBundle/": dial tcp 127.0.0.1:41583: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.001s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:613
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:41583/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:41583
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:546
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:657
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (3) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestInitHandshake366902240
--- FAIL: TestInitHandshake (600.02s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4247596&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4247596&tab=artifacts#/) on master @ [5b871267b62d22854f8c7b92c7c5843100597976](https://github.com/cockroachdb/cockroach/commits/5b871267b62d22854f8c7b92c7c5843100597976):


```
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.001s): context canceled before init bundle received from leader
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.001s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:640
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:548
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:642: -- test log scope end --
--- FAIL: TestInitHandshake (600.07s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4446394&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4446394&tab=artifacts#/) on master @ [67a3e567c6ef9f4fcd58050b003a5706a870a514](https://github.com/cockroachdb/cockroach/commits/67a3e567c6ef9f4fcd58050b003a5706a870a514):


```
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.001s): context canceled before init bundle received from leader
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.001s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:640
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:548
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:642: -- test log scope end --
--- FAIL: TestInitHandshake (600.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4606915&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4606915&tab=artifacts#/) on master @ [9850f2aed2d44e04e08d9f14ff0c98a2bd193ee9](https://github.com/cockroachdb/cockroach/commits/9850f2aed2d44e04e08d9f14ff0c98a2bd193ee9):


```
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.005s): context canceled before init bundle received from leader
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.005s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:640
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (3) operation "init handshake" timed out after 10m0s (took 10m0.003s)
        	            	Wraps: (4) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:548
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (5) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *contextutil.TimeoutError (4) *withstack.withStack (5) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:642: -- test log scope end --
--- FAIL: TestInitHandshake (600.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4621841&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4621841&tab=artifacts#/) on master @ [b3525c95b176f7520173e7a6d3e4a93dca86b198](https://github.com/cockroachdb/cockroach/commits/b3525c95b176f7520173e7a6d3e4a93dca86b198):


```
    init_handshake_test.go:96: n2: generating cert bundle for cluster
    init_handshake_test.go:96: n2: sending cert bundle to peers
    init_handshake_test.go:96: n3: received cert bundle
    init_handshake_test.go:122: 
        	Error Trace:	init_handshake_test.go:122
        	Error:      	Received unexpected error:
        	            	operation "init handshake" timed out after 10m0s (took 10m0.001s): invalid signature in messages from peers; likely due to token mismatch
        	            	(1) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.002s): error when sending bundle to peers as leader: Post "https://127.0.0.1:45741/deliverBundle/": dial tcp 127.0.0.1:45741: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.002s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:615
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:45741/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:45741
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) operation "init handshake" timed out after 10m0s (took 10m0.001s)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:548
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *contextutil.TimeoutError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:642: -- test log scope end --
--- FAIL: TestInitHandshake (600.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4685726&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4685726&tab=artifacts#/) on master @ [090d481f2ec1d1c3e668274f50e4f152b4596a1b](https://github.com/cockroachdb/cockroach/commits/090d481f2ec1d1c3e668274f50e4f152b4596a1b):


```
    init_handshake_test.go:96: n1: trusted peer: 127.0.0.1:38713
    init_handshake_test.go:96: n1: waiting for cert bundle
    init_handshake_test.go:96: n3: sending cert bundle to peers
    init_handshake_test.go:96: n1: received cert bundle
    init_handshake_test.go:122: 
        	Error Trace:	init_handshake_test.go:122
        	Error:      	Received unexpected error:
        	            	invalid signature in messages from peers; likely due to token mismatch
        	            	(1) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.002s): error when sending bundle to peers as leader: Post "https://127.0.0.1:38713/deliverBundle/": dial tcp 127.0.0.1:38713: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.002s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:615
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:38713/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:38713
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:548
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (3) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:642: -- test log scope end --
--- FAIL: TestInitHandshake (600.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78409 server: TestInitHandshake failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4729962&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4729962&tab=artifacts#/) on master @ [bd80d601e6792c9346a0bf04adc7053fc1997ced](https://github.com/cockroachdb/cockroach/commits/bd80d601e6792c9346a0bf04adc7053fc1997ced):


```
    init_handshake_test.go:96: n3: trusted peer: 127.0.0.1:35517
    init_handshake_test.go:96: n3: waiting for cert bundle
    init_handshake_test.go:96: n1: sending cert bundle to peers
    init_handshake_test.go:96: n3: received cert bundle
    init_handshake_test.go:122: 
        	Error Trace:	init_handshake_test.go:122
        	Error:      	Received unexpected error:
        	            	invalid signature in messages from peers; likely due to token mismatch
        	            	(1) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0s (took 10m0.001s): error when sending bundle to peers as leader: Post "https://127.0.0.1:35517/deliverBundle/": dial tcp 127.0.0.1:35517: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0s (took 10m0.001s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:615
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:35517/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:35517
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:548
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:661
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:659
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (3) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:642: -- test log scope end --
--- FAIL: TestInitHandshake (600.02s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78409 server: TestInitHandshake failed [C-test-failure O-robot T-server-and-security branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5384855?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5384855?buildTab=artifacts#/) on master @ [2c08debd6d8c89019f26a3e58cf3b5d3d97b6495](https://github.com/cockroachdb/cockroach/commits/2c08debd6d8c89019f26a3e58cf3b5d3d97b6495):


```
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:32907/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:32907
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0.002s (given timeout 10m0s): context canceled before init bundle received from leader
        	            	  | (1) operation "init handshake" timed out after 10m0.002s (given timeout 10m0s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:641
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:549
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:642: -- test log scope end --
--- FAIL: TestInitHandshake (600.05s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78409 server: TestInitHandshake failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5507642?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5507642?buildTab=artifacts#/) on master @ [8d34ef1ea15850ee1c70470610b6652df4c317de](https://github.com/cockroachdb/cockroach/commits/8d34ef1ea15850ee1c70470610b6652df4c317de):


```
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0.002s (given timeout 10m0s): error when sending bundle to peers as leader: Post "https://127.0.0.1:46715/deliverBundle/": dial tcp 127.0.0.1:46715: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0.002s (given timeout 10m0s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:616
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:46715/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:46715
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:549
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:661: -- test log scope end --
--- FAIL: TestInitHandshake (600.05s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78409 server: TestInitHandshake failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5507641?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5507641?buildTab=artifacts#/) on master @ [8d34ef1ea15850ee1c70470610b6652df4c317de](https://github.com/cockroachdb/cockroach/commits/8d34ef1ea15850ee1c70470610b6652df4c317de):


```
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0.001s (given timeout 10m0s): error when sending bundle to peers as leader: Post "https://127.0.0.1:40301/deliverBundle/": dial tcp 127.0.0.1:40301: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0.001s (given timeout 10m0s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:616
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:40301/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:40301
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:549
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:661: -- test log scope end --
--- FAIL: TestInitHandshake (600.02s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78409 server: TestInitHandshake failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7578695?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7578695?buildTab=artifacts#/) on master @ [a035d2e6e7ea57b30115ecc8ee1a5d553a9e3412](https://github.com/cockroachdb/cockroach/commits/a035d2e6e7ea57b30115ecc8ee1a5d553a9e3412):


```
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0.001s (given timeout 10m0s): error when sending bundle to peers as leader: Post "https://127.0.0.1:38529/deliverBundle/": dial tcp 127.0.0.1:38529: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0.001s (given timeout 10m0s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:616
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:38529/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:38529
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (3) operation "init handshake" timed out after 10m0.001s (given timeout 10m0s)
        	            	Wraps: (4) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:641
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (5) context canceled before init bundle received from leader
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *contextutil.TimeoutError (4) *withstack.withStack (5) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestInitHandshake2447461256
--- FAIL: TestInitHandshake (600.06s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78409 server: TestInitHandshake failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7971744?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7971744?buildTab=artifacts#/) on master @ [d6f98e90684894fd36f53596e6aac355676d232e](https://github.com/cockroachdb/cockroach/commits/d6f98e90684894fd36f53596e6aac355676d232e):


```
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:45821/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:45821
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0.001s (given timeout 10m0s): context canceled before init bundle received from leader
        	            	  | (1) operation "init handshake" timed out after 10m0.001s (given timeout 10m0s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:641
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:549
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestInitHandshake3007958745
--- FAIL: TestInitHandshake (600.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92318 server: TestInitHandshake failed [C-test-failure O-robot T-kv branch-release-22.2]
- #78409 server: TestInitHandshake failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9598999?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9598999?buildTab=artifacts#/) on master @ [8e6cdbd8b3d12ba5d65db101a3abfeb2cb1bf470](https://github.com/cockroachdb/cockroach/commits/8e6cdbd8b3d12ba5d65db101a3abfeb2cb1bf470):


```
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func5
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:101
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_arm64.s:1172
        	            	  | Wraps: (3) context canceled before init bundle received from leader
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	            	Wraps: (2) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0.001s (given timeout 10m0s): error when sending bundle to peers as leader: Post "https://127.0.0.1:37443/deliverBundle/": dial tcp 127.0.0.1:37443: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0.001s (given timeout 10m0s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:616
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_arm64.s:1172
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:37443/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:37443
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:549
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_arm64.s:1172
        	            	Wraps: (4) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *secondary.withSecondaryError (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5dd3194da4351b03928fb98be247cd8e/logTestInitHandshake1136246692
--- FAIL: TestInitHandshake (600.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78409 server: TestInitHandshake failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

server.TestInitHandshake [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9724394?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9724394?buildTab=artifacts#/) on master @ [8124bff03d073f35f4d2b6a2048c7f4417d757d9](https://github.com/cockroachdb/cockroach/commits/8124bff03d073f35f4d2b6a2048c7f4417d757d9):


```
    init_handshake_test.go:96: n3: generating cert bundle for cluster
    init_handshake_test.go:96: n3: sending cert bundle to peers
    init_handshake_test.go:96: n1: received cert bundle
    init_handshake_test.go:122: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:122
        	Error:      	Received unexpected error:
        	            	invalid signature in messages from peers; likely due to token mismatch
        	            	(1) secondary error attachment
        	            	  | operation "init handshake" timed out after 10m0.001s (given timeout 10m0s): error when sending bundle to peers as leader: Post "https://127.0.0.1:35863/deliverBundle/": dial tcp 127.0.0.1:35863: connect: connection refused
        	            	  | (1) operation "init handshake" timed out after 10m0.001s (given timeout 10m0s)
        	            	  | Wraps: (2) attached stack trace
        	            	  |   -- stack trace:
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:616
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  |   | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  |   | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func7
        	            	  |   | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:107
        	            	  |   | runtime.goexit
        	            	  |   | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	  | Wraps: (3) error when sending bundle to peers as leader
        	            	  | Wraps: (4) Post "https://127.0.0.1:35863/deliverBundle/"
        	            	  | Wraps: (5) dial tcp 127.0.0.1:35863
        	            	  | Wraps: (6) connect
        	            	  | Wraps: (7) connection refused
        	            	  | Error types: (1) *contextutil.TimeoutError (2) *withstack.withStack (3) *errutil.withPrefix (4) *url.Error (5) *net.OpError (6) *os.SyscallError (7) syscall.Errno
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/server.initHandshakeHelper
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:549
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:662
        	            	  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
        	            	  | github.com/cockroachdb/cockroach/pkg/server.InitHandshake
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake.go:660
        	            	  | github.com/cockroachdb/cockroach/pkg/server.TestInitHandshake.func6
        	            	  | 	github.com/cockroachdb/cockroach/pkg/server/init_handshake_test.go:104
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (3) invalid signature in messages from peers; likely due to token mismatch
        	            	Error types: (1) *secondary.withSecondaryError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestInitHandshake
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestInitHandshake1741218830
--- FAIL: TestInitHandshake (600.01s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78409 server: TestInitHandshake failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInitHandshake.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/init_handshake.go
+++ b/pkg/server/init_handshake.go
@@ -255,7 +255,12 @@ func (t *tlsInitHandshaker) onTrustInit(
 
 	log.Ops.Infof(ctx, "received valid challenge and CA from: %s", challenge.HostAddress)
 
-	t.trustedPeers <- challenge
+	select {
+	case t.trustedPeers <- challenge:
+	case <-ctx.Done():
+		apiV2InternalError(req.Context(), ctx.Err(), res)
+		return
+	}
 
 	// Acknowledge validation to the client.
 	ack, err := createNodeHostnameAndCA(t.listenAddr, t.tempCerts.CACertificate, t.token)

